Search path environment variables for cross-compilers.  See the discussion
at <http://gcc.gnu.org/ml/gcc/2013-02/msg00124.html>.

--- gcc-4.7.2/gcc/incpath.c	2012-01-27 00:34:58.000000000 +0100
+++ gcc-4.7.2/gcc/incpath.c	2013-02-12 10:11:27.000000000 +0100
@@ -452,7 +452,7 @@ register_include_chains (cpp_reader *pfi
 
   /* CPATH and language-dependent environment variables may add to the
      include chain.  */
-  add_env_var_paths ("CPATH", BRACKET);
+  add_env_var_paths ("CROSS_CPATH", BRACKET);
   add_env_var_paths (lang_env_vars[idx], SYSTEM);
 
   target_c_incpath.extra_pre_includes (sysroot, iprefix, stdinc);

--- gcc-4.7.2/gcc/system.h	2012-02-17 00:16:28.000000000 +0100
+++ gcc-4.7.2/gcc/system.h	2013-02-12 10:22:17.000000000 +0100
@@ -1023,4 +1023,6 @@ helper_const_non_const_cast (const char
 #define DEBUG_VARIABLE
 #endif
 
+#define LIBRARY_PATH_ENV "CROSS_LIBRARY_PATH"
+
 #endif /* ! GCC_SYSTEM_H */

--- gcc-4.7.2/gcc/tlink.c	2012-02-11 09:50:23.000000000 +0100
+++ gcc-4.7.2/gcc/tlink.c	2013-05-23 22:06:19.000000000 +0200
@@ -461,7 +461,7 @@ recompile_files (void)
   file *f;
 
   putenv (xstrdup ("COMPILER_PATH="));
-  putenv (xstrdup ("LIBRARY_PATH="));
+  putenv (xstrdup (LIBRARY_PATH_ENV "="));
 
   while ((f = file_pop ()) != NULL)
     {

--- gcc-4.7.3/gcc/gcc.c	2013-03-08 08:25:09.000000000 +0100
+++ gcc-4.7.3/gcc/gcc.c	2013-05-24 08:58:16.000000000 +0200
@@ -3726,7 +3726,7 @@ process_command (unsigned int decoded_op
     }
 
   temp = getenv (LIBRARY_PATH_ENV);
-  if (temp && *cross_compile == '0')
+  if (temp)
     {
       const char *startp, *endp;
       char *nstore = (char *) alloca (strlen (temp) + 3);
